libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -552,18 +515,22 @@ internal void UpdateOnReceiveAnswer()
         /// <seealso cref="VideoTrackSource.OnDisable"/>
         internal void OnSourceDestroyed()
         {
-            if (_senderTrack is LocalAudioTrack audioTrack)
-            {
-                audioTrack.Dispose();
-            }
-            else if (_senderTrack is LocalVideoTrack videoTrack)
+            // Different from `Source = null`. Don't need to call Source.OnRemovedFromMediaLine
+            // since the Source itself has called this.
+            DestroySenderIfNeeded();
+            UpdateTransceiverDesiredDirection();
+            Source = null;
+        }
+
+        public void OnBeforeSerialize() {}
+
+        public void OnAfterDeserialize()
+        {
+            if (_source)
             {
-                videoTrack.Dispose();
+                // Fill the list of media lines for the source.
+                ((IMediaTrackSourceInternal)_source).OnAddedToMediaLine(this);
```

Comment:
`OnAddedToMediaLine()` asserts that the media line list doesn't contain this source already, but `OnAfterDeserialize()` is called in many places, including apparently when cloning an object, and not necessarily only when the current object is created/loaded the first time (see [docs](https://docs.unity3d.com/ScriptReference/ISerializationCallbackReceiver.html)). So `OnAddedToMediaLine()` should probably be updated to avoid asserting.

Reply:
It shouldn't be called twice on the same object though, right?

Reply:
No idea, the docs seem to say it is...

Reply:
The docs say
> Serialization can occur during all kinds of operations. For example, when using Instantiate() to clone an object, Unity serializes and deserializes the original object

IMHO they seem to suggest to be aware that those callbacks might be called at runtime, in addition to when files are loaded from disk. They don't seem to imply that callbacks might be called arbitrarily on an object that has already been deserialized. So I see no problem here.

Reply:
Fine. "Unity serializes and deserializes the original object" tells me they do are calling it, but on the other hand that makes no sense to deserialize the source object of a clone, so let's assume it's poor wording.

Reply:
I agree it sounds weird but it's reasonable to assume that they mean "serializes the original object and deserializes the serialized original object" 😄